cmd/krew/cmd/list.go

```diff
@@ -24,14 +24,105 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/kubectl/pkg/util/slice"
 
+	"sigs.k8s.io/krew/pkg/constants"
 	"sigs.k8s.io/krew/pkg/installation"
 )
 
+type Entry struct {
+	Name    string
+	Version string
+}
+
+type Entries []Entry
+
+func sortByName(E Entries) Entries {
+	sort.Slice(E, func(a, b int) bool {
+		return E[a].Name < E[b].Name
+	})
+	return E
+}
+
+// Consume produces a junk GroupVersionKind for obj.GetObjectKind().GroupVersionKind().Empty() check to eat in PrintObj()
+type Consume struct {
+	APIVersion string
+	Kind       string
+}
+
+func (c Consume) SetGroupVersionKind(kind schema.GroupVersionKind) {
+	c.APIVersion, c.Kind = kind.ToAPIVersionAndKind()
+}
+
+func (c Consume) GroupVersionKind() schema.GroupVersionKind {
+	return schema.FromAPIVersionAndKind(c.APIVersion, c.Kind)
+}
+
+func (e Entries) GetObjectKind() schema.ObjectKind {
+	return Consume{constants.CurrentAPIVersion, constants.PluginKind}
+}
+
+func (e Entry) GetObjectKind() schema.ObjectKind {
+	return Consume{constants.CurrentAPIVersion, constants.PluginKind}
+}
+
+func (e Entries) DeepCopyObject() runtime.Object {
+	newObj := append(Entries{}, e...)
+	return newObj
+}
+
+func (e Entry) DeepCopyObject() runtime.Object {
+	newObj := Entry{e.Name, e.Version}
+	return newObj
+}
+
+type ListFlags struct {
+	JSONYamlPrintFlags *genericclioptions.JSONYamlPrintFlags
+	OutputFormat       *string
+}
+
+func NewListFlags() *ListFlags {
+	outputFormat := ""
+	return &ListFlags{
+		JSONYamlPrintFlags: genericclioptions.NewJSONYamlPrintFlags(),
+		OutputFormat:       &outputFormat,
+	}
+}
+
+func (f *ListFlags) AllowedFormats() []string {
+	formats := f.JSONYamlPrintFlags.AllowedFormats()
+	return formats
+}
+
+func (f *ListFlags) AddFlags(c *cobra.Command) {}
+
+func (f *ListFlags) ToPrinter() (printers.ResourcePrinter, error) {
+	outputFormat := *f.OutputFormat
+	var printer printers.ResourcePrinter
+
+	switch outputFormat {
+	case "json":
+		printer = &printers.JSONPrinter{}
+	case "yaml":
+		printer = &printers.YAMLPrinter{}
+	default:
+		return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: f.OutputFormat, AllowedFormats: f.AllowedFormats()}
+	}
+
+	return printer, nil
+}
+
 func init() {
+	outputFormat := ""
+	output := NewListFlags()
+
 	// listCmd represents the list command
 	listCmd := &cobra.Command{
-		Use:   "list",
+		Use:   "list [(-o|--output=)json|yaml|wide|name]",
```

Comment:
Cobra already deals with showing possible flags. There is no need to specify those explicitly here.

Reply:
Removed